packages/engine-server/src/drivers/file/noteParser.ts

```diff
@@ -214,21 +218,22 @@ export class NoteParser extends ParserBase {
                 hash: resp.noteHash,
               });
             }
+
+            // Check for duplicate IDs when adding new note to the map
+            if (notesById[notes[0].id] !== undefined) {
```

Comment:
The duplicate note ID check is now only happening for the first note in `notes`, is there any reason for that?

Reply:
`resp.propsList` now returns the note we are trying to create + all the parents that were updated. I wrote this with the idea that only the first note should matter for duplicate check but now that I think about it, its possible that stubs could also have duplicate ids. Let me rework this.